gen2-class-saver-jpeg/main.py

```diff
@@ -0,0 +1,95 @@
+detection_nn = pipeline.createNeuralNetwork()
+detection_nn.setBlobPath(mobilenet_path)
+cam_rgb.preview.link(detection_nn.input)
+
+# Create outputs
+xout_rgb = pipeline.createXLinkOut()
+xout_rgb.setStreamName("rgb")
+cam_rgb.preview.link(xout_rgb.input)
+
+xout_nn = pipeline.createXLinkOut()
+xout_nn.setStreamName("nn")
+detection_nn.out.link(xout_nn.input)
+
+# MobilenetSSD label texts
+texts = ["background", "aeroplane", "bicycle", "bird", "boat", "bottle", "bus", "car", "cat", "chair", "cow",
+         "diningtable", "dog", "horse", "motorbike", "person", "pottedplant", "sheep", "sofa", "train", "tvmonitor"]
+
+for text in texts:
+    (Path(__file__).parent / Path(f'data/{text}')).mkdir(parents=True, exist_ok=True)
+
+
+# Pipeline defined, now the device is connected to
+with dai.Device(pipeline) as device:
+    # Start pipeline
+    device.startPipeline()
+
+    # Output queues will be used to get the rgb frames and nn data from the outputs defined above
+    q_rgb = device.getOutputQueue(name="rgb", maxSize=4, blocking=False)
+    q_nn = device.getOutputQueue(name="nn", maxSize=4, blocking=False)
+
+    frame = None
+    bboxes = []
+    labels = []
+
+    # nn data, being the bounding box locations, are in <0..1> range - they need to be normalized with frame width/height
+    def frame_norm(frame, bbox):
+        return (np.array(bbox) * np.array([*frame.shape[:2], *frame.shape[:2]])[::-1]).astype(int)
+
+
+    while True:
+        # instead of get (blocking) used tryGet (nonblocking) which will return the available data or None otherwise
+        in_rgb = q_rgb.tryGet()
+        in_nn = q_nn.tryGet()
+
+        if in_rgb is not None:
+            # if the data from the rgb camera is available, transform the 1D data into a HxWxC frame
+            shape = (3, in_rgb.getHeight(), in_rgb.getWidth())
+            frame = in_rgb.getData().reshape(shape).transpose(1, 2, 0).astype(np.uint8)
+            frame = np.ascontiguousarray(frame)
+
+        if in_nn is not None:
+            # one detection has 7 numbers, and the last detection is followed by -1 digit, which later is filled with 0
+            bboxes = np.array(in_nn.getFirstLayerFp16())
+            # transform the 1D array into Nx7 matrix
+            bboxes = bboxes.reshape((bboxes.size // 7, 7))
+            # filter out the results which confidence less than a defined threshold
+            bboxes = bboxes[bboxes[:, 2] > 0.5]
+            # Cut bboxes and labels
+            labels = bboxes[:, 1].astype(int)
+            bboxes = bboxes[:, 3:7]
+
+        if frame is not None:
+            # if the frame is available, draw bounding boxes on it and show the frame
+            for raw_bbox, label in zip(bboxes, labels):
+                bbox = frame_norm(frame, raw_bbox)
+                det_frame = frame[bbox[1]:bbox[3], bbox[0]:bbox[2]]
+                cv2.imwrite(f'data/{texts[label]}/{int(time.time() * 10000)}.jpg', det_frame)
```

Comment:
I know you have your own objectives with this code -- but my belief is that people (and I will speak for myself) want to not JUST have the frame of the detected frame that matched -- but I think they want the whole image -- with or without the bounding box / boxes. So potentially saving the raw frame, frame with the bounding box overlaid AND the det_frame would be best. I know this is overkill -- but if this is sample / example code -- I think this would give users the best understanding of what is possible and how to do it.

Reply:
Yes, agreed.  I think we should show saving both the whole image, and the individual ROIs, as then it is best of both worlds.  And just make it clear when it's doing each.

Thanks for the suggestion.  Thoughts @VanDavv ?

Reply:
@wasauce @Luxonis-Brandon Thanks for the feedback, it actually made me push this dataset generation further. I've upgraded the example. Now it stores both raw frames, cropped frames and frames with overlay (bounding boxes and labels). Please see the updated PR description or updated README.md file for more details.

Does that solve your request @wasauce ?

Reply:
Great idea, thank you!  Will try to test today!

Reply:
@VanDavv It certainly does. This is excellent! Thank you.

Reply:
Where can I get this example?

Reply:
@qaqak I've just merged it, you can download it from latest master

Reply:
Thanks @VanDavv